source/common/common/backoff_strategy.cc

```diff
@@ -27,4 +27,26 @@ uint64_t ExponentialBackOffStrategy::computeNextInterval() {
   }
   return current_interval_;
 }
+
+JitteredBackOffStrategy::JitteredBackOffStrategy(uint64_t base_interval,
+                                                 Runtime::RandomGenerator& random)
+    : base_interval_(base_interval), random_(random) {}
+
+JitteredBackOffStrategy::JitteredBackOffStrategy(uint64_t base_interval, uint64_t max_interval,
+                                                 Runtime::RandomGenerator& random)
+    : base_interval_(base_interval), max_interval_(max_interval), random_(random) {
+  ASSERT(base_interval_ < max_interval_);
+}
+
+uint64_t JitteredBackOffStrategy::nextBackOffMs() { return computeNextInterval(); }
+
+void JitteredBackOffStrategy::reset() { current_retry_ = 0; }
+
+uint64_t JitteredBackOffStrategy::computeNextInterval() {
+  current_retry_++;
+  uint32_t multiplier = (1 << current_retry_) - 1;
+  uint64_t new_interval = random_.random() % (base_interval_ * multiplier);
```

Comment:
What are the behavioral tradeoffs between doing what you have here and something more like `(1 << current_retry_) * base_interval + random.random() % some_fixed_interval`? I.e. you still do exponential backoff, but jsut add a small jitter around the point that you backoff to?

I think what you have is fine actually, reading https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/, but it would be worth exploring the tradeoff in the comments here and review.

Reply:
I have compared both approaches, Looks like the one I have has more randomness.
Here is output
(1 << current_retry_) * base_interval + random.random() % some_fixed_interval (fixed interval as 100)
1082
2069
4079
8054
16010
30000

random_.random() % (base_interval_ * (1 << current_retry_) - 1)
102
1065
954
1084
7196
28940
23291
30000
9979